src/components/PopupWithForm.js

```diff
@@ -21,9 +23,11 @@ export default class PopupWithForm extends Popup {
     return this._popupForm;
   }
 
-  // close popup
+  // close popup and reset form along with button text
   close() {
     super.close();
+    this._popupForm.reset();
```

Comment:
You need to **reset** the form only in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.